json/src/spec/state.rs

```diff
@@ -19,38 +19,74 @@
 use std::collections::BTreeMap;
 use crate::{
 	bytes::Bytes,
-	hash::Address,
+	hash::{Address, H256},
 	spec::{Account, Builtin}
 };
 use serde::Deserialize;
 
+/// Recent JSON tests can be either a map or a hash (represented by a string).
+/// See https://github.com/ethereum/tests/issues/637
+#[cfg(any(test, feature = "test-helpers"))]
+#[derive(Clone, Debug, PartialEq, Deserialize)]
+#[serde(untagged)]
+pub enum HashOrMap {
+	/// When the `postState` is large, tests sometimes just include the state root of the last
+	/// successful block here.
+	Hash(H256),
+	/// The expected `postState` of a test
+	Map(BTreeMap<Address, Account>),
+}
+
+/// Recent JSON tests can be either a map or a string. See https://github.com/ethereum/tests/issues/637
+// todo[dvdplm]: what is the point of doing this for non-test code? Needed?
```

Comment:
it is just for the `derive Clone` but let's merge them and have it implement `Clone`

Reply:
You mean like this?

```rust
pub enum HashOrMap {
  …
}

#[cfg(any(test, feature = "test-helpers"))]
impl Clone for HashOrMap {
  …
}
```

Reply:
I mean't just remove the conditional compliation and derive Clone